app/src/main/java/org/akvo/flow/activity/AddUserActivity.java

```diff
@@ -61,6 +63,10 @@ public void onClick(View v) {
                 db.savePreference(ConstantUtil.DEVICE_IDENT_KEY, deviceId);
                 db.close();
 
+                // Trigger the SurveyDownload Service, in order to make
+                // a backend connection with the new Device ID
```

Comment:
I guess it could be moved to where SurveyActivity receives the return value from AddUserActivity. I was sticking as close as possible to the similar code in PreferencesActivity.